Allow selecting scenes to color correct by grid

ddohler opened the pull request:
## Overview

This implements part of the color correction wireframes which show an interface for selecting scenes by clicking grid cells displayed on the map. Because we don't need to show a count of scenes like we do on the browse page, I was able to get away with generating the grid by drawing a border around the Leaflet tiles.

There is also some general cleanup, including removing an unused hook from the `colorCorrectScenes` component, moving the logic for getting all of a Project's Scenes into the Project service, and fixing an incorrect query parameter in the `getProjectSceneCount` method that was causing it to unnecessarily return a full page of results. 

### Checklist

- [ ] ~Styleguide updated, if necessary~
- [ ] ~Swagger specification updated, if necessary~

### Demo

![gridselect](https://cloud.githubusercontent.com/assets/447977/21204234/5f1ac466-c224-11e6-904d-1ef97bc0c9c2.png)

## Testing Instructions

 * Navigate to the color correction interface for a project containing the Egypt scenes.
 * Zoom in until the scenes are spread across more than one grid cell.
 * Click on a grid cell, verify that some scenes are selected.
 * Switch into color correction mode and verify that the selected scenes overlap with the clicked grid cell.

Closes #727

ddohler commented:
I've updated this to show a highlight when a grid cell is clicked, and to allow toggling the selection by clicking in the same grid cell again. I've also fixed up a race condition where you could click between two grid cells really quickly and end up with the scenes from the first-clicked grid cell rather than the second-clicked one.
![gridselecthighlight](https://cloud.githubusercontent.com/assets/447977/21243117/6551d8d0-c2e5-11e6-9388-9b674895a45d.png)

I think it would be best to hold off on making the grid cells smaller for now; I think the best way to do it would be to create a custom `GridLayer` with a smaller tile size, but that would require a deeper level of interaction with the map than we can easily support with the current structure. The current cell sizes are the same as what we're using on the browse page (i.e. quad-tree tiles) so they're consistent at least.

notthatbreezy commented:
Sounds good about holding off on the cell size - we'll have to put it in next sprint probably.

The PR looks good and the interactions are fixed. Everything works, but I am getting a JS error in the console when I use your branch:

![js-error](https://cloud.githubusercontent.com/assets/898060/21264373/f103bbda-c369-11e6-9c6b-500ed7ec057b.png)

ddohler commented:
Yup I noticed that error when I got started on rebasing last night and fixed it. I've pushed up a rebased and fixed version. The map is re-broken now because I took out the bandage for that to avoid conflicting with #786 , but this should be ready to go otherwise.
